docs/guides/nextjs-setup.md

```diff
@@ -0,0 +1,285 @@
+
+## Start a new Next.js React App
+
+### Create the Next.js app
+
+```tsx
+npx create-next-app
+```
+Provide the following answers when prompted:
+- What is your project named? `react-sdk-walkthrough`
+- Would you like to use TypeScript? `Yes`
+- Would you like to use ESLint? `Yes`
+- Would you like to use Tailwind CSS? `No`
+- Would you like to use `src/` directory? `No`
+- Would you like to use App Router? (recommended) `Yes`
+- Would you like to customize the default import alias (@/*)? `No`
+
+### Run the Next.js app
+
+```tsx
+cd react-sdk-walkthrough
+yarn dev
+```
+
+Open a browser to the location output for `local` (e.g., [http://localhost:3000](http://localhost:3000/))
+
+## Eppo flag setup
+1. [Create and copy an SDK key](/sdks/sdk-keys.md) in your Eppo app if you don't already have one.
+2. [Create a feature flag](/quick-starts/feature-flag-quickstart.md/#2-create-a-flag) in your Eppo app.
+3. Make sure your flag is turned on the environment that correlates to your SDK key.
+
+## Client rendered component setup
+
+### Install the Eppo JS client SDK
+
+```bash
+yarn add @eppo/js-client-sdk
+```
+
+### Create a randomization provider
+
+1. Create a provider based off our [Usage in React](/sdks/client-sdks/javascript#usage-in-react) documentation
+2. Create a new `/app` file called `EppoRandomizationProvider.tsx`
+3. Copy and paste the following example code into `EppoRandomizationProvider.tsx`:
+```tsx
+import { ReactElement, useEffect, useState } from "react";
+
+import { init } from "@eppo/js-client-sdk";
+
+interface IEppoRandomizationProvider {
+  waitForInitialization?: boolean;
+  children: ReactElement;
+  loadingComponent?: ReactElement;
+}
+
+const EppoRandomizationProvider = ({
+  waitForInitialization = true,
+  children,
+  loadingComponent = <div>Loading...</div>,
+}: IEppoRandomizationProvider): ReactElement => {
+  const [isInitialized, setIsInitialized] = useState(false);
+  useEffect(() => {
+    init({
+      apiKey: "<SDK-KEY>",
+      assignmentLogger: {
+        logAssignment(assignment) {
+          console.log('TODO: log', assignment)
+        },
+      },
+    }).then(() => {
+      return setIsInitialized(true);
+    });
+  }, []);
+
+  if (!waitForInitialization || isInitialized) {
+    return children;
+  }
+  return loadingComponent;
+}
+
+export default EppoRandomizationProvider;
+```
+4. Replace API key placeholder with your SDK key
+
+### Create a component to use the randomization provider
+1. Create a new `/app` file  `OfferExperiment.tsx`
+2. Copy and paste the following code into the `OfferExperiment.tsx` file.
+```tsx
+import { useMemo } from "react";
+import { getInstance } from "@eppo/js-client-sdk";
+
+const OfferComponent = () => {
+
+    const subjectKey = "<SUBJECT-KEY>"
+
+    const assignedVariation = useMemo(() => {
+      const eppoClient = getInstance();
+      return eppoClient.getStringAssignment("<FLAG-KEY>", subjectKey, <SUBJECT-ATTRIBUTES>, "<DEFAULT-VALUE>");
```

Comment:
I find the inconsistency a bit jarring, maybe do this instead?
```suggestion
      return eppoClient.getStringAssignment(<FLAG-KEY>, subjectKey, <SUBJECT-ATTRIBUTES>, <DEFAULT-VALUE>);
```

Reply:
Yea, I copied/pasted from other examples in our docs. I think the difference is conveying this value is a string and this value is not a string. I agree though, maybe all the string quotes should be removed and people can reference the schema for the method.